app/components/map-interface/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { makeStyles, useTheme } from "@rneui/themed"
+import MapView, { Region } from "react-native-maps"
+import MapStyles from "./map-styles.json"
+import React, { useRef } from "react"
+import { BusinessMapMarkersQuery } from "@app/graphql/generated"
+import { useI18nContext } from "@app/i18n/i18n-react"
+import MapMarker from "../map-marker"
+
+type Props = {
+  data?: BusinessMapMarkersQuery
+  userLocation?: Region
+  handleMapPress: () => void
+  handleMarkerPress: (_: MarkerData) => void
+  focusedMarker: MarkerData | null
+  handleCalloutPress: (_: MarkerData) => void
+}
+
+export type MarkerData = {
+  readonly __typename: "MapMarker"
+  readonly username?: string | null | undefined
+  readonly mapInfo: {
+    readonly __typename: "MapInfo"
+    readonly title: string
+    readonly coordinates: {
+      readonly __typename: "Coordinates"
+      readonly longitude: number
+      readonly latitude: number
+    }
+  }
+}
+
+export default function MapInterface({
+  data,
+  userLocation,
+  handleMapPress,
+  handleMarkerPress,
+  focusedMarker,
+  handleCalloutPress,
+}: Props) {
+  const {
+    theme: { colors, mode: themeMode },
+  } = useTheme()
+  const styles = useStyles()
+  const { LL } = useI18nContext()
+  const text = React.useMemo(() => LL.MapScreen.payBusiness(), [LL.MapScreen])
+
+  const mapViewRef = useRef<MapView>(null)
+
+  return (
+    <MapView
+      ref={mapViewRef}
+      style={styles.map}
+      showsUserLocation={true}
+      initialRegion={userLocation}
+      customMapStyle={themeMode === "dark" ? MapStyles.dark : MapStyles.light}
+      onPress={handleMapPress}
+    >
+      {(data?.businessMapMarkers ?? []).reduce(
+        (arr: React.ReactElement[], item: MarkerData | null) => {
+          if (item?.username) {
+            arr.push(
+              <MapMarker
+                key={item.username}
+                item={item}
+                color={colors._orange}
+                handleCalloutPress={handleCalloutPress}
+                handleMarkerPress={handleMarkerPress}
+                isFocused={focusedMarker?.username === item.username}
+                text={text}
+                styles={styles}
+              />,
+            )
+          }
+
+          return arr
+        },
+        [] as React.ReactElement[],
```

Comment:
is casting necessary? can't we have a `const empty: React.ReactElement[] = []` instead?

Reply:
I dunno - I was always told that was good form. It's to show the arrays got a type from initialisation even when it's empty. I'll remove it.